lib/managers/files.js

```diff
@@ -968,6 +991,230 @@ Files.prototype.deleteVersion = function(fileID, versionID, callback) {
+			},
+			json: false,
+			body: part
+		};
+
+	this.client.put(apiURL, params, function(err, response) {
+
+		if (err) {
+			callback(err);
+			return;
+		}
+
+		if (response.statusCode !== 200) {
+			callback(errors.buildUnexpectedResponseError(response));
+			return;
+		}
+
+		// @TODO: When API passes back chunk object, return to user directly
+		callback(null, {
+			part_id: partID,
+			offset: offset,
+			size: part.length
+		});
+	});
+};
+
+/**
+ * Commit an upload session after all parts have been uploaded, creating the new file
+ *
+ * API Endpoint: '/files/upload-sessions/:sessionID/commit'
+ * Method: POST
+ *
+ * @param {string} sessionID - The ID of the upload session to commit
+ * @param {string} fileHash - The base64-encoded SHA-1 hash of the file being uploaded
+ * @param {UploadPart[]} parts - The list of uploaded chunks to be committed
+ * @param {?Object} options - Optional parameters, can be left null
+ * @param {Function} callback - Passed the new file information if successful
+ * @returns {void}
+ */
+Files.prototype.commitUploadSession = function(sessionID, fileHash, parts, options, callback) {
+
+	var apiURL = this.client._uploadBaseURL + urlPath(BASE_PATH, UPLOAD_SESSION_SUBRESOURCE, sessionID, 'commit'),
+		params = {
+			headers: {
+				Digest: 'SHA=' + fileHash
+			},
+			body: {
+				// @TODO: if parts was not passed in, get it from the API
+				parts,
+				attributes: options
+			}
+		};
+
+	var self = this;
+	this.client.post(apiURL, params, function(err, response) {
+
+		if (err) {
+			callback(err);
+			return;
+		}
+
+		if (response.statusCode === 201) {
+			callback(null, response.body);
+			return;
+		}
+
+		if (response.statusCode === 202) {
+			var retryInterval = response.headers['retry-after'] || 1;
+			setTimeout(function() {
+				self.commitUploadSession(sessionID, fileHash, parts, options, callback);
+			}, retryInterval * 1000);
+			return;
+		}
+
+		callback(errors.buildUnexpectedResponseError(response));
+	});
+};
+
+/**
+ * Abort an upload session, discarding any chunks that were uploaded to it
+ *
+ * API Endpoint: '/files/upload-sessions/:sessionID'
+ * Method: DELETE
+ *
+ * @param {string} sessionID - The ID of the upload session to commit
+ * @param {Function} callback - Passed nothing if successful, error otherwise
+ * @returns {void}
+ */
+Files.prototype.abortUploadSession = function(sessionID, callback) {
+
+	var apiURL = this.client._uploadBaseURL + urlPath(BASE_PATH, UPLOAD_SESSION_SUBRESOURCE, sessionID);
+
+	this.client.del(apiURL, null, this.client.defaultResponseHandler(callback));
+};
+
+/**
+ * Get a list of all parts that have been uploaded to an upload session
+ *
+ * API Endpoint: '/files/upload-session/:sessionID/parts'
+ * Method: GET
+ *
+ * @param {string} sessionID - The ID of the session to get a list of parts from
+ * @param {?Object} options - Optional parameters, can be left null
+ * @param {string} [options.marker] - Paging marker for the list of parts
```

Comment:
Yeah, they should.  That's all documented in the [API docs](https://docs.box.com/reference#marker-based-paging)